tree/ntuple/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -27,6 +28,54 @@
 #include <string>
 #include <type_traits>
 
+#ifndef R__LITTLE_ENDIAN
```

Comment:
We have the `R__BYTESWAP` macro in RConfig.hxx, which is defined for little-endian (probably because most other machine-independent stuff in ROOT is represented big-endian).
I'm not sure, though, if we should rely on this one or just move these lines to RConfig.hxx and use `R__LITTLE_ENDIAN`.

Reply:
Also, it seems that `R__BYTESWAP` is currently defined unconditionally in RConfig.hxx (that requires `#undef` on our unit tests).  Any preference and/or ideas?

Reply:
I'd say moving `R__LITTLE_ENDIAN` to `RConfig.hxx` is the best approach.

Reply:
Having two different preprocessor macros for the same thing would be confusing. Unless you are planning to change the name everywhere, sticking with R__BYTESWAP is preferred. (It is also used in PR #10308).

> Also, it seems that `R__BYTESWAP` is currently defined unconditionally in RConfig.hxx (that requires `#undef` on our unit tests). Any preference and/or ideas?

No - it is conditionally defined for little endian architectures, and undefined otherwise, e.g. it is not defined for the linux on s390x build:

https://github.com/root-project/root/blob/3d5ecb5a31b428a0659d6525835da686145fee11/core/foundation/inc/ROOT/RConfig.hxx#L223-L228

Reply:
`#if R__LITTLE_ENDIAN == 1`  →  `#ifdef R__BYTESWAP`
`#if R__LITTLE_ENDIAN == 0`  →  `#ifndef R__BYTESWAP`

Reply:
> Doesn't the definition of `RByteSwap` belongs inside the `Byteswap.h` ? (or a new version there of in `core/foundation/inc/ROOT/`) ?

It might, yes.  @jblomer, anything against?